common/functions/src/aggregates/aggregator_common.rs

```diff
@@ -89,16 +89,30 @@ pub fn eval_aggr(
     let cols: Vec<ColumnRef> = columns.iter().map(|c| c.column().clone()).collect();
 
     let func = factory.get(name, params, arguments)?;
+    let data_type = func.return_type()?;
 
     let arena = Bump::new();
     let place = arena.alloc_layout(func.state_layout());
     let addr = place.into();
     func.init_state(addr);
-    func.accumulate(addr, &cols, None, rows)?;
 
-    let data_type = func.return_type()?;
-    let mut builder = data_type.create_mutable(1024);
-    func.merge_result(addr, builder.as_mut())?;
+    let f = func.clone();
+    // we need a temporary function to catch the errors
+    let apply = || -> Result<ColumnRef> {
+        func.accumulate(addr, &cols, None, rows)?;
+        let mut builder = data_type.create_mutable(1024);
+        func.merge_result(addr, builder.as_mut())?;
+
+        Ok(builder.to_column())
+    };
 
-    Ok(builder.to_column())
+    let result = apply();
+
+    if f.need_manual_drop_state() {
```

Comment:
need `exit_guard`

Reply:
`apply` function is used to catch the whole errors.

`exit_guard ` can not catch the `Bump`.

Reply:
when panic?

Reply:
I think `exit_guard`  can't catch the panic either.

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a80480b9211961b420a2ce55f0f58cf2